File: 4139-minimum-absolute-distance-between-mirror-pairs/minimum-absolute-distance-between-mirror-pairs.py
class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        def reverse_num(x: int) -> int:
            rev = 0
            while x > 0:
                rev = rev * 10 + (x % 10)
                x //= 10
            return rev

        last_seen: dict[int, int] = {}
        ans = float("inf")
        for i, current in enumerate(nums):
            if current in last_seen:
                ans = min(ans, i - last_seen[current])
            last_seen[reverse_num(current)] = i
        return -1 if ans == float("inf") else ans

File: 4139-minimum-absolute-distance-between-mirror-pairs/minimum-absolute-distance-between-mirror-pairs.js
var minMirrorPairDistance = function (nums) {
    const reverseNum = (x) => {
        let rev = 0
        while (x > 0) {
            rev = rev * 10 + (x % 10)
            x = Math.floor(x / 10)
        }
        return rev
    }
    const lastSeen = new Map()
    let ans = Infinity
    for (let i = 0; i < nums.length; i++) {
        const current = nums[i]
        if (lastSeen.has(current)) {
            ans = Math.min(ans, i - lastSeen.get(current))
        }
        lastSeen.set(reverseNum(current), i)
    }
    return ans === Infinity ? -1 : ans
};